src/psbt.cpp

```diff
@@ -265,7 +271,7 @@ void PSBTOutput::Merge(const PSBTOutput& output)
     if (redeem_script.empty() && !output.redeem_script.empty()) redeem_script = output.redeem_script;
     if (witness_script.empty() && !output.witness_script.empty()) witness_script = output.witness_script;
     if (m_tap_internal_key.IsNull() && !output.m_tap_internal_key.IsNull()) m_tap_internal_key = output.m_tap_internal_key;
-    if (m_tap_tree.has_value() && !output.m_tap_tree.has_value()) m_tap_tree = output.m_tap_tree;
+    if (!m_tap_tree.empty() && output.m_tap_tree.empty()) m_tap_tree = output.m_tap_tree;
```

Comment:
isn't this the opposite of what we want?

why is the logic switched to "clear this m_tap_tree if merging output has none set" from "if unset, set m_tap_tree if output has one set".

Reply:
The logic is still "clear this m_tap_tree if merging output has none set". It's just that "has value" is the opposite of "is empty" so it had to be inversed.

Reply:
hmmm, even if not inverted, is the original behavior the intended behavior and why?

Other lines seem to be opposite, e.g. 273 has `set m_tap_key_internal from non-null output.m_tap_key_internal` semantics, not `unset m_tap_key_internal from null output.m_tap_key_internal`

```
if (m_tap_internal_key.IsNull() && !output.m_tap_internal_key.IsNull()) m_tap_internal_key = output.m_tap_internal_key;
```

Reply:
There should be a test covering this stuff as well?

Reply:
Added a test.